packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -933,41 +940,152 @@ class DropdownButtonFormField<T> extends FormField<T> {
   /// The [DropdownButton] [items] parameters must not be null.
   DropdownButtonFormField({
     Key key,
-    T value,
-    @required List<DropdownMenuItem<T>> items,
+    this.value,
+    @required this.items,
+    this.hint,
     this.onChanged,
-    InputDecoration decoration = const InputDecoration(),
+    this.decoration = const InputDecoration(),
     FormFieldSetter<T> onSaved,
     FormFieldValidator<T> validator,
-    Widget hint,
+    bool autovalidate = false,
+    this.disabledHint,
+    this.elevation = 8,
+    this.style,
+    this.icon,
+    this.iconDisabledColor,
+    this.iconEnabledColor,
+    this.iconSize = 24.0,
+    this.isDense = false,
+    this.isExpanded = false,
   }) : assert(decoration != null),
        super(
          key: key,
          onSaved: onSaved,
          initialValue: value,
          validator: validator,
+         autovalidate: autovalidate,
          builder: (FormFieldState<T> field) {
-           final InputDecoration effectiveDecoration = decoration
-             .applyDefaults(Theme.of(field.context).inputDecorationTheme);
+           final InputDecoration effectiveDecoration = decoration.applyDefaults(
+             Theme.of(field.context).inputDecorationTheme,
+           );
            return InputDecorator(
              decoration: effectiveDecoration.copyWith(errorText: field.errorText),
              isEmpty: value == null,
              child: DropdownButtonHideUnderline(
                child: DropdownButton<T>(
-                 isDense: true,
                  value: value,
                  items: items,
                  hint: hint,
                  onChanged: field.didChange,
+                 disabledHint: disabledHint,
+                 elevation: elevation,
+                 style: style,
+                 icon: icon,
+                 iconDisabledColor: iconDisabledColor,
+                 iconEnabledColor: iconEnabledColor,
+                 iconSize: iconSize,
+                 isDense: isDense,
+                 isExpanded: isExpanded,
                ),
              ),
            );
          }
        );
 
+  /// The list of items the user can select.
+  ///
+  /// If the [onChanged] callback is null or the list of items is null
+  /// then the dropdown button will be disabled, i.e. its arrow will be
+  /// displayed in grey and it will not respond to input. A disabled button
+  /// will display the [disabledHint] widget if it is non-null.
+  final List<DropdownMenuItem<T>> items;
+
+  /// The value of the currently selected [DropdownMenuItem], or null if no
+  /// item has been selected. If `value` is null then the menu is popped up as
+  /// if the first item were selected.
+  final T value;
+
+  /// A placeholder widget that is displayed if no item is selected.
+  final Widget hint;
+
+  /// A message to show when the dropdown is disabled.
+  ///
+  /// Displayed if [items] is null.
```

Comment:
I saw that you updated this to not include [onChanged] being set to `null`  -- Could you make the default behavior the same as with a regular `DropdownButton` and ensure all docs reflect the same behavior? I believe the intended behavior is that if [onChanged] is null, the `disabledHint` should be shown.

ie. the [DropdownButtonFormField.onChanged] API docs still says that "A disabled button will display the [disabledHint] widget if it is non-null.", which is what the [DropdownButton.onChanged] API docs say.

Reply:
I'm aware that `DropdownButtonFormField` passes `field.didChanged` instead of the 
passed in `onChanged` callback. See if you can align the behavior here with `DropdownButton` instead so that developers aren't confused by mismatching experiences between widgets.

Reply:
Done